pyteal/ast/abi/util_test.py

```diff
@@ -10,289 +10,350 @@
+    TypeAnnotationTest(
+        annotation=abi.Tuple3,
+        expected=TypeError,
+    ),
+    TypeAnnotationTest(
+        annotation=abi.Tuple3[
+            abi.Tuple1[abi.Uint32],
+            abi.StaticArray[abi.Bool, Literal[55]],
+            abi.Tuple2[abi.Uint32, abi.Uint16],
+        ],
+        expected=abi.TupleTypeSpec(
+            abi.TupleTypeSpec(abi.Uint32TypeSpec()),
+            abi.StaticArrayTypeSpec(abi.BoolTypeSpec(), 55),
+            abi.TupleTypeSpec(abi.Uint32TypeSpec(), abi.Uint16TypeSpec()),
         ),
-        TypeAnnotationTest(
-            annotation=abi.StaticBytes[Literal[10]],
-            expected=abi.StaticBytesTypeSpec(10),
+    ),
+    TypeAnnotationTest(
+        annotation=abi.Tuple4[abi.Uint32, abi.Uint16, abi.Byte, abi.Bool],
+        expected=abi.TupleTypeSpec(
+            abi.Uint32TypeSpec(),
+            abi.Uint16TypeSpec(),
+            abi.ByteTypeSpec(),
+            abi.BoolTypeSpec(),
         ),
-        TypeAnnotationTest(
-            annotation=abi.DynamicBytes,
-            expected=abi.DynamicBytesTypeSpec(),
+    ),
+    TypeAnnotationTest(
+        annotation=abi.Tuple4,
+        expected=TypeError,
+    ),
+    TypeAnnotationTest(
+        annotation=abi.Tuple5[abi.Uint32, abi.Uint16, abi.Byte, abi.Bool, abi.Tuple0],
+        expected=abi.TupleTypeSpec(
+            abi.Uint32TypeSpec(),
+            abi.Uint16TypeSpec(),
+            abi.ByteTypeSpec(),
+            abi.BoolTypeSpec(),
+            abi.TupleTypeSpec(),
         ),
-    ]
+    ),
+    TypeAnnotationTest(
+        annotation=abi.Tuple5,
+        expected=TypeError,
+    ),
+    TypeAnnotationTest(
+        annotation=ExampleNamedTuple, expected=ExampleNamedTuple().type_spec()
+    ),
+    TypeAnnotationTest(
+        annotation=list[abi.Uint16],
+        expected=TypeError,
+    ),
+    TypeAnnotationTest(annotation=abi.Transaction, expected=abi.TransactionTypeSpec()),
+    TypeAnnotationTest(
+        annotation=abi.PaymentTransaction, expected=abi.PaymentTransactionTypeSpec()
+    ),
+    TypeAnnotationTest(
+        annotation=abi.KeyRegisterTransaction,
+        expected=abi.KeyRegisterTransactionTypeSpec(),
+    ),
+    TypeAnnotationTest(
+        annotation=abi.AssetConfigTransaction,
+        expected=abi.AssetConfigTransactionTypeSpec(),
+    ),
+    TypeAnnotationTest(
+        annotation=abi.AssetFreezeTransaction,
+        expected=abi.AssetFreezeTransactionTypeSpec(),
+    ),
+    TypeAnnotationTest(
+        annotation=abi.AssetTransferTransaction,
+        expected=abi.AssetTransferTransactionTypeSpec(),
+    ),
+    TypeAnnotationTest(
+        annotation=abi.ApplicationCallTransaction,
+        expected=abi.ApplicationCallTransactionTypeSpec(),
+    ),
+    TypeAnnotationTest(annotation=abi.Account, expected=abi.AccountTypeSpec()),
+    TypeAnnotationTest(annotation=abi.Asset, expected=abi.AssetTypeSpec()),
+    TypeAnnotationTest(annotation=abi.Application, expected=abi.ApplicationTypeSpec()),
```

Comment:
The fact that I also changed the format of test cases makes this this hard to see (sorry about that), but these are the only new cases.

The `DynamicBytes` and `StaticBytes` cases were reordered to be closer to the dynamic and static array tests, but are otherwise unchanged.

Reply:
After making the changes in cb1d8e3f57ca74b64a1d2e5dec9e277a185760c8, I discovered test cases for `String` and `Address` were also missing, so I've added them as well.